data/transactions/verify/txn.go

```diff
@@ -314,3 +314,113 @@ func LogicSig(txn *transactions.SignedTxn, ctx *Context) error {
 	return nil
 
 }
+
+// PaysetGroups verifies that the payset have a good signature and that the underlying
+// transactions are properly constructed.
+// Note that this does not check whether a payset is valid against the ledger:
+// a PaysetGroups may be well-formed, but a payset might contain an overspend.
+//
+// This version of verify is performing the verification over the provided execution pool.
+func PaysetGroups(ctx context.Context, payset [][]transactions.SignedTxn, blk bookkeeping.Block, verificationPool execpool.BacklogPool) (err error) {
+	txnGroupIdx := 0
+	const txnPerGroupThreshold = 32
+	proto, ok := config.Consensus[blk.BlockHeader.CurrentProtocol]
+	if !ok {
+		return protocol.Error(blk.BlockHeader.CurrentProtocol)
+	}
+	if len(payset) == 0 {
+		return nil
+	}
+	spec := transactions.SpecialAddresses{
+		FeeSink:     blk.BlockHeader.FeeSink,
+		RewardsPool: blk.BlockHeader.RewardsPool,
+	}
+	var txnGroups [][]transactions.SignedTxn
+	// prepare up to 16 concurrent worksets.
+	worksets := make(chan struct{}, 16)
+	worksDoneCh := make(chan interface{}, 16)
```

Comment:
why 16?

Reply:
I'll add const to define these with some explanation.